pkgs/misc/vscode-registries/vscode-marketplace/default.nix

```diff
@@ -0,0 +1,99 @@
+{ lib
+, vscode-registry-commons
+, callPackage
+, overlays ? [ ]
+}:
+
+with vscode-registry-commons.registry-lib;
+
+let
+
+  registry-reference-list-fetched = lib.importJSON ./registry-reference-list.json;
+
+  base = final:
+    {
+      cacheUrl = "https://vscode.blob.core.windows.net/gallery/index";
```

Comment:
Could it be that this url is heavily outdated or needs some parameters as https://vscode.blob.core.windows.net/gallery/index shows a heavily outdated version for e.g. https://marketplace.visualstudio.com/items?itemName=fterrag.vscode-php-cs-fixer ?

Reply:
What a bad news.

Now the revelant APIs for the VSCode Marketplace are only `serviceUrl` and `itemUrl`, whics are also what the VSCode-like editors use query the extensions. I haven't found any publicly available documentation about the query, but Open VSX has already implemented [a server-side relay]{https://github.com/eclipse/openvsx/wiki/Using-Open-VSX-in-VS-Code). Maybe they have already figured out the trick.

Reply:
I think it is more of a legal issue and we are only allowed to support adding the extensions to the unmodified vscode to my limited understanding. The website https://marketplace.visualstudio.com/vscode contains "Download extension" links that at least show the link for individual extensions. The question is where you can get a list of extensions

Reply:
> I think it is more of a legal issue and we are only allowed to support adding the extensions to the unmodified vscode to my limited understanding.
That's why I open this PR. (The original purpose is to switch to Open VSX, and later superherontj suggests that it make more sense to support multiple registries)


The open-vsx people seems to know the trick, but neither they nor the Microsoft has published any documentation about the technical detail.